src/features/import/hooks/usePreflight.ts

```diff
@@ -0,0 +1,227 @@
+import { CountryCode } from 'libphonenumber-js';
+import { useState } from 'react';
+
+import { ALERT_STATUS } from '../components/ImportDialog/Steps/ImportAlert';
+import { importResultAdd } from '../store';
+import messageIds from '../l10n/messageIds';
+import prepareImportOperations from '../utils/prepareImportOperations';
+import useFieldTitle from '../../../utils/hooks/useFieldTitle';
+import { useMessages } from 'core/i18n';
+import useOrganization from 'features/organizations/hooks/useOrganization';
+import useOrganizations from 'features/organizations/hooks/useOrganizations';
+import useTags from 'features/tags/hooks/useTags';
+import { ZetkinTag } from 'utils/types/zetkin';
+import getAddedOrgsSummary, {
+  checkAllValuesAreZero,
+  checkEmptyObj,
+} from '../utils/getAddedOrgsSummary';
+import {
+  IMPORT_ERROR,
+  PersonImport,
+  ZetkinPersonImportPostBody,
+} from '../utils/types';
+import { useApiClient, useAppDispatch, useAppSelector } from 'core/hooks';
+
+export interface Alert {
+  msg: string;
+  status: ALERT_STATUS;
+  title: string;
+}
+
+export default function usePreflight(orgId: number) {
```

Comment:
There is something "wrong" here, where an import that previews to 0 changes generates an error. It's probably correct to interrupt, but the error message seems like an "undefined error" and would probably be easier to understand if it said something like "your import wouldn't cause any changes, so you can't proceed" (or something nicer that I can't think of right now).

![image](https://github.com/zetkin/app.zetkin.org/assets/550212/88377b51-4e73-4d9f-a63d-961f2b8ca4a3)

Reply:
Agree! I remember this was on my mind from the beginning, but slipped. It is just the message that it was in the design, and I agree it should be more specifically saying why the import "fails"